### PR TITLE
Verification that the already existing OSDs are not restarted post add capacity (bug 1931601)

### LINKED ISSUE #3893: add_capacity test: add verification that the already existing OSDs are not restarted post add capacity

### PR DESCRIPTION
**Pull request by ebenahar**
Fixes #3893

Signed-off-by: Elad Ben Aharon <ebenahar@redhat.com>

### CODE REVIEW

File: tests/manage/z_cluster/cluster_expansion/test_add_capacity.py [hunk 1]
```diff
@@ -18,16 +19,34 @@
...
+    restarted_osds = list()
+    logger.info(
+        "Checking if existing OSD pods were restarted post add capacity (bug 1931601)"
+    )
+    for pod in existing_osd_pod_names:
+        if pod.name not in osd_pods_post_expansion:
+            restarted_osds.append(pod.name)
+    assert (
+        len(restarted_osds) == 0
+    ), f"The following OSD pods were restarted post add capacity: {[pod.name for pod in restarted_osds]}"
```
Comment (jilju) [inline tests/manage/z_cluster/cluster_expansion/test_add_capacity.py:48]: List ```existing_osd_pod_names``` contain the names of OSDs
```suggestion
    for pod in existing_osd_pod_names:
        if pod not in osd_pods_post_expansion:
            restarted_osds.append(pod)
    assert (
        len(restarted_osds) == 0
    ), f"The following OSD pods were restarted post add capacity: {restarted_osds}"
```

**Review by jilju — APPROVED** [review]

File: tests/manage/z_cluster/cluster_expansion/test_add_capacity.py (hunk 1 above)
Comment (ebenahar) [reply tests/manage/z_cluster/cluster_expansion/test_add_capacity.py]: I think I fixed it now

**Review by jilju — APPROVED** [review]

File: tests/manage/z_cluster/cluster_expansion/test_add_capacity.py [hunk 2]
```diff
@@ -18,16 +19,36 @@
...
 
+logger = logging.getLogger(__name__)
+
+
 def add_capacity_test():
     osd_size = storage_cluster.get_osd_size()
+    existing_osd_pods = get_osd_pods()
+    existing_osd_pod_names = [pod.name for pod in existing_osd_pods]
     result = storage_cluster.add_capacity(osd_size)
+    osd_pods_post_expansion = get_osd_pods()
```
Comment (hnallurv) [inline tests/manage/z_cluster/cluster_expansion/test_add_capacity.py:38]: There is a generic function to get the restart count for all the pods in a given namespace in [here](https://github.com/red-hat-storage/ocs-ci/blob/048dd21117f968becea32c15144cf7f5ce26035f/ocs_ci/ocs/resources/pod.py#L1531). We could use that for not just checking the restarts of osd pod but also other pods...
Usage of the above function can be seen [here](https://github.com/red-hat-storage/ocs-ci/blob/master/tests/manage/z_cluster/cluster_expansion/test_add_capacity_entry_exit_criteria.py#L212)

Comment (ebenahar) [reply tests/manage/z_cluster/cluster_expansion/test_add_capacity.py:38]: restart count is not what we are looking for. We are looking for OSD pods deletion

Comment (ebenahar) [reply tests/manage/z_cluster/cluster_expansion/test_add_capacity.py:38]: I updated the logging with this information. thanks for noticing

**Review by jilju — APPROVED** [review]

File: tests/manage/z_cluster/cluster_expansion/test_add_capacity.py [hunk 3]
```diff
@@ -18,16 +19,36 @@
...
+    osd_pod_names_post_expansion = [pod.name for pod in osd_pods_post_expansion]
```
Comment (PrasadDesala) [inline tests/manage/z_cluster/cluster_expansion/test_add_capacity.py:39]: add_capacity()  seems like is updating the device count and returning it.
After add-capacity are we waiting for the new osd pods to reach running state? If not there are chances that osd_pod_names_post_expansion list  contains only the osd's before expansion.

Comment (ebenahar) [reply tests/manage/z_cluster/cluster_expansion/test_add_capacity.py:39]: the intention is the examine the already existing OSDs, not the newly created ones. Hence, it doesn't matter if the new OSDs were created already or not

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**